fastlane/lib/fastlane/actions/cocoapods.rb

```diff
@@ -14,7 +14,7 @@ def self.run(params)
           cmd << ["cd '#{podfile_folder}' &&"]
         end
 
-        cmd << ['bundle exec'] if File.exist?('Gemfile') && params[:use_bundle_exec]
+        cmd << ['bundle exec'] if params[:use_bundle_exec] && shell_out_should_use_bundle_exec?
```

Comment:
Although I'm willing to conceded that that would be the cherry  on top, more or less, so if the `contained_fastlane?` thing is fixed I'd be okay to merge.